botocore/client.py

```diff
@@ -534,7 +539,7 @@ def _make_api_call(self, operation_name, api_params):
             http, parsed_response = event_response
         else:
             http, parsed_response = self._endpoint.make_request(
-                operation_model, request_dict)
+                operation_model, request_dict, request_context)
```

Comment:
I am not a fan of adding the request_context to the ``make_request`` signature because we would have to modify all of the other methods of the ``Endpoint`` class. What if we did the following:

1) Keep the signature the same for all of the ``endpoint`` module
2) Introduce the idea of passing a context to ``prepare_request_dict``: https://github.com/boto/botocore/blob/develop/botocore/awsrequest.py#L263
3) Add the context to AWSRequest object from the request dict when we create it: https://github.com/boto/botocore/blob/develop/botocore/awsrequest.py#L294.

Another thing that I am not sure if I am liking is the fact we take a context that we expose in the all of the events related to parameter building and strap it into a context living on ``AWSRequest`` that we have never exposed. It seems that there is a chance for really hard to determine bugs. I would prefer that the user could never affect the context of ``AWSRequest`` because that could affect behavior that we rely on.